lib/improver/threshold.py

```diff
@@ -171,16 +173,36 @@ def __init__(self, thresholds, fuzzy_factor=None,
             assert bounds[0] <= thr, bounds_msg
             assert bounds[1] >= thr, bounds_msg
 
-        self.below_thresh_ok = below_thresh_ok
+        # Lists of known logical comparisons. The final entry in each list
+        # must be recognisable by eval()
+        self.inequality_dict = {}
+        self.inequality_dict.update(dict.fromkeys(['ge', 'GE', '>='],
+                                                  {'function': operator.ge,
+                                                   'spp_string': 'above',
+                                                   'valid_for_fuzzy': False}))
+        self.inequality_dict.update(dict.fromkeys(['gt', 'GT', '>'],
+                                                  {'function': operator.gt,
+                                                   'spp_string': 'above',
+                                                   'valid_for_fuzzy': True}))
+        self.inequality_dict.update(dict.fromkeys(['le', 'LE', '<='],
+                                                  {'function': operator.le,
+                                                   'spp_string': 'below',
+                                                   'valid_for_fuzzy': False}))
+        self.inequality_dict.update(dict.fromkeys(['lt', 'LT', '<'],
+                                                  {'function': operator.lt,
+                                                   'spp_string': 'below',
+                                                   'valid_for_fuzzy': True}))
```

Comment:
The logic of the first reviewer was that a user asking for a fuzzy threshold and also <= or >= was probably not thinking clearly. Plus, if Bruce decides that we need different meta-data to indicate the equality logic, then we wouldn't want to include this for fuzzy thresholds.

Reply:
I'm not sure the metadata question applies.  In that case there would be different metadata to describe fuzzily thresholded outputs, which you would condition on first eg:
```
if fuzzy_factor:
   # some metadata
elif "spp_string" == ...:
   # some other metadata
# etc
```
But in any case, I think this is complexity that we don't need to support at this point - and if we need it in the future, it would be better to add complexity when we know the function of it, than to include generic complexity at the start that might not turn out to support what we need.

Reply:
Code logic changed so that when calling threshold with fuzzy-thresholds active, identical behaviour is seen with either > or >= and < or <=. The discussed error has been removed. The doc-strings for "comparison-operator" have been updated to explicitly state this.